### PR TITLE
ref: add `read_timeout` and `connect_timeout` config options (`remote add`)

### PR DESCRIPTION
**Pull request by dtrifiro**
- remote modify: remove duplicated config options for s3
- remote modify: add timeout config options for http remote

Related: 
- iterative/dvc#8722
- iterative/dvc-http#29

### CODE REVIEW

File: content/docs/command-reference/remote/modify.md [hunk 1]
```diff
@@ -1162,6 +1146,22 @@ by HDFS. Read more about by expanding the WebHDFS section in
...
+  ```cli
+  $ dvc remote modify myremote read_timeout 300
+  ```
+
+- `connect_timeout` - set the time in seconds till a timeout exception is thrown
+  when attempting to make a connection (60 by default). Let's set it to 5
+  minutes for example:
+
+  ```cli
+  $ dvc remote modify myremote connect_timeout 300
```
Comment (jorgeorpinel) [inline content/docs/command-reference/remote/modify.md:1162]: Forgot to change this
````suggestion
  $ dvc remote add myremote read_timeout 300
  ```

- `connect_timeout` - set the time in seconds till a timeout exception is thrown
  when attempting to make a connection (60 by default). Let's set it to 5
  minutes for example:

  ```cli
  $ dvc remote add myremote connect_timeout 300
````

**Review by jorgeorpinel — APPROVED** [review]

File: content/docs/command-reference/remote/modify.md [hunk 2]
```diff
@@ -1162,6 +1146,22 @@ by HDFS. Read more about by expanding the WebHDFS section in
   $ dvc remote modify myremote ssl_verify path/to/ca_bundle.pem
   ```
 
+- `read_timeout` - set the time in seconds till a timeout exception is thrown
+  when attempting to read from a connection (60 by default). Let's set it to 5
+  minutes for example:
+
+  ```cli
+  $ dvc remote add myremote read_timeout 300
```
Comment (dtrifiro) [inline content/docs/command-reference/remote/modify.md:1154]: @jorgeorpinel I'm not sure I understand this change, in order to change a parameter we need to use `modify`, not `add`. Is it something else that you wanted to modify?

File: content/docs/command-reference/remote/modify.md [hunk 3]
```diff
@@ -1162,6 +1146,22 @@ by HDFS. Read more about by expanding the WebHDFS section in
   $ dvc remote modify myremote ssl_verify path/to/ca_bundle.pem
   ```
 
+- `read_timeout` - set the time in seconds till a timeout exception is thrown
+  when attempting to read from a connection (60 by default). Let's set it to 5
```
Comment (dtrifiro) [inline content/docs/command-reference/remote/modify.md:1150]:
```suggestion
  when attempting to read a portion of data from a connection (60 by default). Let's set it to 5
```

might be clearer

**Comment by dberenbaum** [comment]
By the way @dtrifiro, unless there's a specific reason you tagged @jorgeorpinel, let's pick someone else from the dev team to review, or use @iterative/docs if there's nobody that's suitable, so we don't depend on @jorgeorpinel to be the primary reviewer for dvc docs. He will still review later if he thinks there's something important to mention.

**Comment by dberenbaum** [comment]
Should we link to https://docs.aiohttp.org/en/stable/client_reference.html?highlight=clienttimeout#clienttimeout and mention which args these map to? Also, sorry that I'm late on this, but why not name them `sock_connect` and `sock_read` to match aiohttp?

**Comment by dtrifiro** [comment]
@dberenbaum I think that's not really necessary, and it does not map 1-1. Regarding this and naming choices, see this discussion: https://github.com/iterative/dvc/pull/8722#discussion_r1056730462

**Review by dberenbaum — APPROVED** [review]

File: content/docs/command-reference/remote/modify.md (hunk 2 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/remote/modify.md]: Sorry for the late reply here @dtrifiro! Yup, that was a typo on my end, thanks for fixing.

**Comment by jorgeorpinel** [comment]
Specifically for cmd/API ref updates, unless they seem related to releases or something like moving info that should be in GS or UG ([example](https://github.com/iterative/dvc.org/pull/4264/files#r1094049854)), indeed I am not usually able to respond in a timely manner, sry.

**Pull request merged**